nucleus/deploy/client.py

```diff
@@ -113,9 +150,13 @@ def create_model_endpoint(
         env_params: Dict[str, str],
         requirements: Optional[List[str]] = None,
         gpu_type: Optional[str] = None,
+        aws_role: Optional[str] = None,
```

Comment:
I think this is leaking too many implementation details. Showing people that we use s3 for signed URL uploads should be fine because they're signed (though maybe we still want to think long term about how we accept data).

Maybe this was already a security hole and this change is only highlighting the fact, but doesn't this mean that I could modify the client library and pass in my AWS role of choice to Scale's hosted Deploy service?

I would suggest going a bit more abstract and following a DI (dependency injection)-esque approach. You can pass in an optional "AuthDecorator" object/function, where external users pass `None` (the default), and internal users pass in something that lives in our internal repo that enriches this payload with the role info. This way, you don't expose as many details about AWS, which makes our cleanup later easier. Even this isn't the best abstraction, but it'd still be better IMO.

But we'd still need to address the underlying security issue. I would suggest that we also do server-side validation, where we strip out the `aws_role` and `results_s3_bucket` in the payload in the Gateway service only if a scaleapi-auth'ed `userid` is present (i.e. it's an external user). @syandroo thoughts?

Reply:
Discussed offline - I misunderstood and forgot that we are forcing the right `aws_role` in `scaleapi` server-side.